### DISCUSSION TITLE
Make a simple Animation

### DISCUSSION THREAD

**Discussion by HarrisonLeece** [alandefreitas/matplotplusplus#209] [Q&A] [answered]
How can I create a simple animation from my plots?  I just need to eliminate the press ENTER to continue tag and add a for loop, but I can't find anyway to skip the press enter to continue. Thoughts?

Thanks

**Comment by alandefreitas** [comment, marked as answer]
Use `draw()` instead of `show()`

**Reply by HarrisonLeece** [reply]
When I compile it says "draw();" is not defined in scope.  I didn't change the location of the line I replaced.

**Reply by alandefreitas** [reply]
If you are in quiet mode, you can look for the `draw` function (standalone or in the axes_object).
Or just use the `save` (standalone or in axes_object) functions as you plot.

Only `show()` requires you to press enter. It's intended to block execution before the program quits.
Everything else should work fine as long as you don't use `show()` in the middle of a loop.

A few examples, like [this one](https://github.com/alandefreitas/matplotplusplus/blob/master/examples/geography/geoplot/geoplot_7.cpp), are animations.

**Reply by HarrisonLeece** [reply]
I was looking through the source code, and saw your name on everything I looked at :eyes: so thank you for working so hard on this package.

This being my second C++ project, I am struggling a bit with the some basic tasks.  I have #included "matplot/matplot.h"  Is this insufficient for the standalone draw() function to be included?  What is quiet mode?  Google didn't immediately bring up a good answer.

You mentioned that there is a draw function which is stand alone, but your example only included the draw function as part of the axes object.  I am really hoping for this to be an easy fix since this is the last step in this part of the project and I am tired.

Edit;  It wasnt so bad.

I copied the function I already had , changed its name and added a new argument 
`void animated_single_channel(vector<int> channel, auto anifig) {...}`
inside the function just call plot, xlabel, ylabel, title as normal.  Instead of `show();` use `anifig->draw();`

In your main, create a matplot figure object
`auto animationFigure = matplot::figure();`
And pass this figure object into your new function.
